### ISSUE TITLE
v5 endpoints cannot receive messages sent by v6 SendOnly endpoints

### ISSUE THREAD

**Issue by mikeminutillo** [Particular/NServiceBus.AzureStorageQueues#181]
## Who's affected
All users sending messages from V6 SendOnly endpoints to V5 endpoints.

## Symptoms
Message sent from V6 SendOnly endpoint will cause exception when processed by receiving V5 endpoint and ultimately result in message being moved to the `error` queue.

## Details 
This problem is caused by the fact that V6 SendOnly endpoint does not add `ReplyToAddress` to it's outgoing message headers and the V5 receiver is expecting a non-null ReplyToAddress (see [here](https://github.com/Particular/NServiceBus.AzureStorageQueues/blob/support-6.2/src/Transport/AzureMessageQueueReceiver.cs#L184)).

NOTE: A V5 SendOnly endpoint *is* able to send messages to a V5 receiver. This is because the ReplyToAddress gets filled in [here](https://github.com/Particular/NServiceBus.AzureStorageQueues/blob/master-6.2.1-bump-up/src/Transport/AzureMessageQueueSender.cs#L111)

_Edit by @boblangley_
Raised from Customer case 00023775

**Comment by mikeminutillo** [comment]
Repro is https://github.com/mikeminutillo/Asq181Repro

Note https://github.com/mikeminutillo/Asq181Repro/blob/master/SenderV6/Program.cs#L22

**Comment by Scooletz** [comment]
Investigation:

I followed this issue and checked how the `ReplyTo` header was appended in v5 Core. It followed the same principles as in v6 with not appending this header for `SendOnly` endpoints (it was done not by a pipeline behavior, still resulting in the same outcome). Next, I checked this header is applied for ASQ in v6 and there it's [done conditionally](https://github.com/Particular/NServiceBus.AzureStorageQueues/blob/master/src/Transport/Dispatcher.cs#L119)

```
var toSend = new MessageWrapper
{
    //...
    ReplyToAddress = headers.GetValueOrDefault(Headers.ReplyToAddress),
    //...
};
```

which is different from the `v5` that was [done unconditionally](https://github.com/Particular/NServiceBus.AzureStorageQueues/blob/support-6.2/src/Transport/AzureMessageQueueSender.cs#L111):

```
var replyToAddress = validation.Determine(config.Settings, message.ReplyToAddress ??
        options.ReplyToAddress ?? config.LocalAddress, config.TransportConnectionString());
```

I think that we could tackle it either by adding this value unconditionally, which does not fix existing messages, or fixing `v5` to allow messages with an empty `ReplyTo`.
